internal/pkg/cli/cli.go

```diff
@@ -107,3 +107,19 @@ func relativeDockerfilePath(ws copilotDirGetter, dockerfilePath string) (string,
 	}
 	return relDfPath, nil
 }
+
+// dfBuildRequired returns if the container image should be built from local Dockerfile.
+func dfBuildRequired(svc interface{}) (bool, error) {
+	type manifest interface {
+		BuildRequired() (bool, error)
+	}
+	mf, ok := svc.(manifest)
+	if !ok {
+		return false, fmt.Errorf("service does not have required methods BuildRequired()")
+	}
```

Comment:
honestly i prefer this to return an error, though ideally this error should never be triggered. However, the benefit is when someone edits this or adds new type that calls `dfBuildRequired()`, it will be easier for debugging instead of swallowing the error and just returns `false`.